akka-docs/src/main/paradox/typed/persistence-snapshot.md

```diff
@@ -0,0 +1,64 @@
+# Persistence - snapshotting
+
+## Snapshots
+
+As you model your domain using @ref:[persistent actors](persistence.md), you may notice that some actors may be
+prone to accumulating extremely long event logs and experiencing long recovery times. Sometimes, the right approach
+may be to split out into a set of shorter lived actors. However, when this is not an option, you can use snapshots
+to reduce recovery times drastically.
+
+Persistent actors can save snapshots of internal state every N events or when a given predicated of the state
+is fulfilled.
+
+Scala
+:  @@snip [BasicPersistentActorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #snapshottingEveryN }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshottingEveryN }
+
+
+Scala
+:  @@snip [BasicPersistentActorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #snapshottingPredicate }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshottingPredicate }
+
+When a snapshot is triggered incoming commands are stashed until the snapshot has been saved. This means that
+the state can safely be mutable although the serialization and storage of the state is performed asynchronously,
+since the state instance will not be updated by new events until after the snapshot has been saved.
+
+During recovery, the persistent actor is using the previously saved snapshot to initialize the state and then events
```

Comment:
When we say 'previously saved' do we mean if the `keep` is say set to 2, the system snapshots 2 batches plus the fresh one. And 'using' this most recent of that?

Reply:
changed "previously saved" to "latest saved", "fresh" also refers to "latest" so changed that as well